managedplugin/plugin.go

```diff
@@ -180,11 +180,15 @@ func (c *Client) downloadPlugin(ctx context.Context, typ PluginType) error {
 		org, name := pathSplit[0], pathSplit[1]
 		c.LocalPath = filepath.Join(c.directory, "plugins", typ.String(), org, name, c.config.Version, "plugin")
 		c.LocalPath = WithBinarySuffix(c.LocalPath)
-		return DownloadPluginFromGithub(ctx, c.logger, c.LocalPath, org, name, c.config.Version, typ, dops)
+		downloadMode, err := DownloadPluginFromGithub(ctx, c.logger, c.LocalPath, org, name, c.config.Version, typ, dops)
+		c.metrics.DownloadMode = downloadMode
+		return err
 	case RegistryDocker:
 		if imageAvailable, err := isDockerImageAvailable(ctx, c.config.Path); err != nil {
+			c.metrics.DownloadMode = DownloadModeCached
```

Comment:
I think in this switch case `DownloadModeCached` is only set in the error case. It should also be set to `DownloadModeCached` outside of this if?

Reply:
I changed the logic so that the `AssetSource` is only set in metrics when the plugin download does not fail.